ntbs-service/Pages/Notifications/NotificationModelBase.cs

```diff
@@ -10,27 +10,36 @@ namespace ntbs_service.Pages_Notifications
     public abstract class NotificationModelBase : PageModel
     {
         protected INotificationService service;
+        protected IAuthorizationService authorizationService;
 
-        public NotificationModelBase(INotificationService service) 
+        public NotificationModelBase(INotificationService service, IAuthorizationService authorizationService)
         {
             this.service = service;
+            this.authorizationService = authorizationService;
         }
 
         protected NotificationGroup Group;
-
         public int NumberOfLinkedNotifications { get; set; }
 
         public Notification Notification { get; set; }
-
         public NotificationBannerModel NotificationBannerModel { get; set; }
 
+        [BindProperty]
+        public bool HasEditPermission { get; set; }
+
         [BindProperty]
         public int NotificationId { get; set; }
 
+        protected async Task AuthorizeAndSetBannerAsync()
+        {
+            HasEditPermission = await authorizationService.CanEdit(User, Notification);
+            NotificationBannerModel = new NotificationBannerModel(Notification, HasEditPermission);
+        }
+
         protected async Task GetLinkedNotifications()
         {
             Group = await GetNotificationGroupAsync();
-            NumberOfLinkedNotifications = Group != null ? (Group.Notifications.Count -1) : 0;
+            NumberOfLinkedNotifications = Group != null ? ( Group.Notifications.Count - 1 ) : 0;

```

Comment:
Tried to use the newfangled suggestion here to make it easy, but decided it would probably be harder.

Not introduced in your changeset but I think this syntax is cleaner:
`NumberOfLinkedNotifications = Group?.Notifications.Count - 1 ?? 0;`